ionoscloud/resource_server_cube.go

```diff
@@ -777,10 +797,35 @@ func resourceCubeServerRead(ctx context.Context, d *schema.ResourceData, meta in
 
 func resourceCubeServerUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(services.SdkBundle).CloudApiClient
+	ss := cloudapiserver.Service{Client: client, Meta: meta, D: d}
 
 	dcId := d.Get("datacenter_id").(string)
 	request := ionoscloud.ServerProperties{}
 
+	currentVmState, err := ss.GetVmState(ctx, dcId, d.Id())
+	if err != nil {
+		diags := diag.FromErr(fmt.Errorf("could not retrieve server vmState: %w", err))
+		return diags
+	}
+	if strings.EqualFold(currentVmState, cloudapiserver.CubeVMStateStop) && !d.HasChange("vm_state") {
+		diags := diag.FromErr(fmt.Errorf("cannot update a suspended Cube Server, must change the state to RUNNING first"))
+		return diags
+	}
+
+	// Unsuspend a Cube server first, before applying other changes
+	if d.HasChange("vm_state") && strings.EqualFold(currentVmState, cloudapiserver.CubeVMStateStop) {
+		_, newVmState := d.GetChange("vm_state")
+		if !strings.EqualFold(newVmState.(string), cloudapiserver.VMStateStart) {
+			diags := diag.FromErr(fmt.Errorf("cannot update a suspended Cube Server, must change the state to RUNNING first"))
```

Comment:
again, best to print newVmState here. Also might be best to change a bit the err message to not confuse it with the one above